### DISCUSSION TITLE
Attribute/Group Code field mask

### DISCUSSION THREAD

**Discussion by paultvis** [atrocore/atrocore#358] [Q&A]
Hi,

We have just installed Atro and wanted to start importing data from another system. When I imported the attribute groups errors were received stating that some of the values failed validation. The error is 'The value of the field "code" does not correspond to the pattern "/^[a-z_0-9{}]+$/".' I looked in entities for attribute codes, and the code field does not exist so I am assuming that it is some kind of system field (although not all entities seem to have it). Our attributes and group names have underscores in them, so where can I change the validation mask to allow underscores?

Thanks

Paul

**Comment by atrolex** [comment]
Hi Paul, good point..  In the (over) next release we will implement, that codes will be generated automatically, it not provided via API. I think it will solve your problems.

**Reply by paultvis** [reply]
Hi Alex. Thanks. Does that mean for now there is no way to change the mask?

**Reply by paultvis** [reply]
And that in the future the code will not be shown?

**Reply by atrolex** [reply]
Сurrently there is no way to change the mask quickly. You could implement your own module for that..
In the future code will be optional, and will be shown, if configured to be shown via Adminstration / Layouts. The value will be generated automatically, if not provided or provided as empty.

**Reply by exeqtwasnotfree** [reply]
Hi, is this related to my problem what importing an update of products? It fails due to GTIN is not unique. The product is in the system, using SKU as identifier, but for some reason import fails because not unique.

**Reply by paultvis** [reply]
@atrolex, are the masks applied to the labels also? All errors returned from the job state that code has failed the mask validation, but looking at the import date we have something like code=some_value and label=[some_value]. We have entries where _ has been processed so that seems OK.

**Reply by atrolex** [reply]
I am not quite sure, I understand you properly. There are no labels for attributes. Only Name and Description. There are no limit for these.

**Reply by paultvis** [reply]
Sorry, I was a bit quick - they were failing the mast as there were caps in the text.

**Reply by atrolex** [reply]
@exeqtwasnotfree i think if GTIN (MPN, SKU) is provided it should be unique within certain catalog. In case of GTIN, MPN, SKU it is hardcorded. While importing the system is searching for a data record using your identifiers, if more than 1 record is found, you will get an error. Verify your configuration.
